AC3-2-Final-Retake-Fireblog/AC3-2-Final-Retake-Fireblog/FeedViewController.swift

```diff
@@ -0,0 +1,220 @@
+//
+//
+//  Created by Erica Y Stevens on 4/13/17.
+//
+//
+
+import UIKit
+import Firebase
+
+class FeedViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    // MARK: Stored Properties
+    
+    fileprivate var _authHandle: FIRAuthStateDidChangeListenerHandle!
+    fileprivate var _refHandle: FIRDatabaseHandle!
+    fileprivate var _storageHandle: FIRStorageHandle!
+    var databaseReference: FIRDatabaseReference!
+    var storageReference: FIRStorageReference!
+    var user: FIRUser?
+    var postInfoFromDatabase = [String:[String:Any]]()
+    var posts = [Post]()
+    
+    // MARK: View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .white
+        
+        checkForLoggedInUser()
+        setupViewHiearchy()
+        configureConstraints()
+        configureTableView()
+        getPostsFromDatabase { (postInfoDict) in
+            DispatchQueue.main.async {
+                self.buildPostObjectsFromDatabaseInfo(postInfoDict, completion: { (post) in
+                    self.posts.append(post)
+                    self.tableView.reloadData()
+                })
+            }
+        }
+    }
+    
+    // MARK: View Hierarchy and Constraints
+    
+    func setupViewHiearchy() {
+        self.view.addSubview(tableView)
+    }
+    
+    func configureConstraints() {
+        let _ = [
+            tableView.widthAnchor.constraint(equalTo: self.view.widthAnchor),
+            tableView.heightAnchor.constraint(equalTo: self.view.heightAnchor),
+            tableView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),
+            tableView.centerYAnchor.constraint(equalTo: self.view.centerYAnchor)
+            ].map{ $0.isActive = true }
+    }
+    
+    // MARK: TableView Configuration
+    
+    func configureTableView() {
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        self.tableView.register(FeedTableViewCell.self, forCellReuseIdentifier: "FeedCell")
+        self.tableView.estimatedRowHeight = 100
+        self.tableView.rowHeight = UITableViewAutomaticDimension
+        self.tableView.separatorStyle = .none
+    }
+    
+    // MARK: Firebase
+    
+    func checkForLoggedInUser() {
+        _authHandle = FIRAuth.auth()?.addStateDidChangeListener({ (auth: FIRAuth?, user: FIRUser?) in
+            if let activeUser = user {
+                self.user = activeUser
+                self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Logout", style: .plain, target: self, action: #selector(self.logOutUser(_:)))
+            }
+        })
+    }
+    
+    func logOutUser(_ sender: UIBarButtonItem) {
+        if FIRAuth.auth()?.currentUser != nil {
+            do {
+                try FIRAuth.auth()?.signOut()
+                let loginController = LoginRegistrationViewController()
+                self.present(loginController, animated: true, completion: nil)
```

Comment:
Similarly, the listener in AppDelegate also presents the login when this state changes. Did you test whether that would have displayed the login when the state changes after logout?